src/components/TaskItem/TaskItem.tsx

```diff
@@ -0,0 +1,75 @@
+import styles from "./TaskItem.module.scss";
+import Checkbox from "../Checkbox/Checkbox";
+import { useEffect, useRef, useState } from "react";
+import { MdDelete } from "react-icons/md";
+import { AiFillEdit } from "react-icons/ai";
+import IconButton from "../IconButton/IconButton";
+import clsx from "clsx";
+
+type TaskItemProps = {
+  message: string;
+  completed: boolean;
+  messageEditable: boolean;
+};
+
+const TaskItem = ({
+  message = "first task",
+  completed = false,
+  messageEditable = false,
+}: TaskItemProps) => {
+  const [title, setTitle] = useState(message);
+  const [checked, setChecked] = useState(completed);
+  const [isEditable, setIsEditbale] = useState(messageEditable);
+  const ref = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (ref?.current && isEditable) {
+      ref.current.focus();
+    }
+  }, [isEditable]);
+
+  return (
+    <div className={styles.TaskItem}>
+      <Checkbox checked={checked} onChange={() => setChecked(!checked)} />
+      <input
+        ref={ref}
+        className={clsx(styles.taskInput, {
+          [styles.taskInputEditable]: isEditable && !checked,
+          [styles.taskInputCompleted]: checked,
+        })}
+        type="text"
+        value={title}
+        onChange={(e) => setTitle(e.target.value)}
+        readOnly={checked}
+        onClick={() => {
+          setIsEditbale(true);
+        }}
+        onBlur={(e) => {
```

Comment:
do not make an anonymous function if its not simple () => setState()
Instead in this cases extract this in `handleOnInputBlur = (e) => ....`